# **CHI-SQUARE**

In [29]:
from google.colab import files
uploaded=files.upload()

Saving Book1.xlsx to Book1 (2).xlsx


In [30]:
import pandas as pd
import numpy as np
from scipy.stats import chi2_contingency
from scipy import stats
data=pd.read_excel('Book1.xlsx',index_col=0)
data

Smart Thermostat  Smart Light  Total
Satisfaction                                          
Very Satisfied                  50           70    120
Satisfied                       80          100    180
Neutral                         60           90    150
Unsatisfied                     30           50     80
Very Unsatisfied                20           50     70
Total                          240          360    600

In [ ]:
#NULL HYPOTHESIS=THERE IS NO SIGNIFICANT ASSOCIATION BETWEEN THE TYPE OF SMART HOME DEVICE PURCHASED AND CUSTOMER SATISFACTION LEVEL
#ALTERNATE HYPOTHESIS=THERE IS SIGNIFICANT ASSOCIATION BETWEEN THE TYPE OF SMART HOME DEVICE PURCHASED AND CUSTOMER SATISFACTION LEVEL

In [31]:
observed=np.array([[50,70],
                   [80,100],
                   [60,90],
                   [30,50],
                   [20,50]])
observed

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [32]:
#CHI SQUARE STATISTIC VALUE=5.638
#DF=4
#P_VALUE=0.227
chi2_contingency(observed)

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

In [33]:
#CHI SQUARE CRITICAL VALUE=9.487
#SIGNIFICANCE LEVEL(ALPHA)=0.05
#AS WE CAN SEE CHI SQUARE STATISTIC VALUE < CHI SQUARE CRITICAL VALUE
#SO, NULL HYPOTHESIS IS TRUE, THERE IS NO SIGNIFICANT ASSOCIATION BETWEEN THE SMART HOME DEVICE PURCHASED AND CUSTOMER SATISFACTION LEVEL
stats.chi2.ppf(1-0.05,4)

9.487729036781154

# **HYPOTHESIS TESTING**

In [1]:
import numpy as np
from scipy import stats

In [ ]:
# NULL HYPOTHESIS=THE MEAN WEEKLY COST OPERATING IS NOT HIGHER THAN THE THEORETICAL MEAN
# ALTERNATE HYPOTHESIS=THE MEAN WEEKLY COST OPERATING IS HIGHER THAN THE THEORETICAL MEAN

In [ ]:
# SAMPLE MEAN = 3050
# STD DEV = 25
# SAMPLE SIZE = 25
# THEORETICAL MEAN = (1000 + 5 * 600)=4000            # (X=600)
# ALPHA VALUE = 0.05

In [13]:
# FIRST CALCULATE THE TEST STATISTIC
# t=(sample mean - theoretical mean)/(std dev/np.sqrt(sample size))
t=(3050 - 4000)/(25/np.sqrt(25))
t

-190.0

In [14]:
# FIND THE CRITICAL VALUE
# DF=SAMPLE SIZE-1
c_value=stats.t.ppf(1-0.05,24)
c_value

1.7108820799094275

AS WE CAN SEE T-STATISTIC < T- CRITICAL VALUE [-190 < 1.71] SO, WE FAIL TO REJECT NULL HYPOTHESIS.

CONCLUSION= THE RESTAURANT OWNER CLAIM WAS CORRECT.